### ISSUE TITLE
Drop outdated PHP versions on stable branch

### ISSUE THREAD

**Issue by core23** [sonata-project/dev-kit#232]
I want to start a new discussion (3rd attempt....) about dropping outdated PHP versions on the stable branch, because we stumple opon some issues with newer third party dependencies, e.g. HTTPlug which requires at least PHP 5.5.

According to semver dropping a minor depenency is no BC break at all. Also the feedback on my [twitter poll](https://twitter.com/core23de/status/817044245018185728) showed a similar response about supported depencies.

Could we drop the minor PHP versions on the stable branch or can we just allow failures for outdated PHP versions.

Refs 
https://github.com/sonata-project/SonataAdminBundle/pull/4276#discussion_r96574850
https://github.com/sonata-project/SonataSeoBundle/pull/161#issuecomment-277433232

**Comment by greg0ire** [comment]
I think we should amend our policy so that things that are no longer supported (no security fixes, or even no bugfixes) can be dropped on the stable branch.

**Comment by core23** [comment]
Maybe start a new poll on twitter for that?

**Comment by greg0ire** [comment]
The other poll you made about php was similar, I think it's safe to assume our userbase is ok with this.

**Comment by Soullivaneuh** [comment]
Personal opinion: Go f*ck*ng ahead! I'm already on 7.0!

In a more general way, we can't consider php as a simple dependencies, because it's a system dependency IMHO.

Plus, we can't forgot we decide to maintain major branches only (2.x, 3.x etc...). So if we push, for example a v3.12 release with php `^5.6` instead of `^5.3` and we found a big on v3.11, we won't be able to deliver a bug fix on a php 5.3 compatible release.

And no, we can't manage minor branches like Symfony IMHO, it would be overkill. We are not a big company and we don't have time for that.

In conclusion, it could be interesting to find or ask why Symfony do maintain php constraint on minor version instead of lot of other projects, like doctrine for example.

I's a :-1: for me ATM, especially because of bugfixes releases. Waiting more answers/information.

cc @rande

**Comment by Soullivaneuh** [comment]
> The other poll you made about php was similar, I think it's safe to assume our userbase is ok with this.

I disagree with that. IMHO it's a big difference between major and minor releases.

**Comment by Soullivaneuh** [comment]
Addition : Even if we decide to do this, we can't drop a major version of a dependency on a minor release, even for PHP or composer.

**Comment by greg0ire** [comment]
Good point about the poll, that's indeed not the same thing. Regarding this issue, I still think we should be able dropping software that is no longer maintained. Otherwise, we might simply be stuck with a bug that can't be fixed.

**Comment by greg0ire** [comment]
That's why we must release major versions more often. Way more often.

**Comment by core23** [comment]
I'll close this for now. We should keep a note about dropping PHP minor / major versions for stable branches in our docs. https://github.com/sonata-project/dev-kit/issues/236

**Comment by greg0ire** [comment]
> And no, we can't manage minor branches like Symfony IMHO, it would be overkill. We are not a big company and we don't have time for that.

@Soullivaneuh : we are going to have a 3-branch system though, is that right?

**Comment by greg0ire** [comment]
A big thing to take into account is that "outdated" means "not maintained anymore". As a small team, I think we can't really afford to support that kind of thing.
See http://php.net/supported-versions.php, IMO we should not support versions that appear in red on this page.

**Comment by OskarStark** [comment]
> IMO we should not support versions that appear in red on this page.

👍

**Comment by jordisala1991** [comment]
We dropped php up to 5.6 on stable on all repositories.

Closing as this is solved.
